test/services/test_wiki_compiler.py

```
@@ -0,0 +1,856 @@
+        from cli_agent_orchestrator.clients.database import Base
+        from cli_agent_orchestrator.services.memory_service import MemoryService
+
+        engine = create_engine("sqlite:///:memory:", connect_args={"check_same_thread": False})
+        Base.metadata.create_all(bind=engine)
+        svc = MemoryService(base_dir=None, db_engine=engine)  # base_dir picked by service
+        ctx = {
+            "terminal_id": "t1",
+            "session_name": "s1",
+            "agent_profile": "developer",
+            "provider": "claude_code",
+            "cwd": "/tmp/testproj",
```

Ruff (0.15.17)

[error] Probable insecure usage of temporary file or directory: "/tmp/testproj"

(S108)